pyne/alara.py

```diff
@@ -890,5 +893,365 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+    fluxes = np.array([])
+    for m in range(num_mats):
+        for n in range(num_n_groups):
+            temp_spectrum = np.zeros(num_n_groups)
+            temp_spectrum[n] = neutron_spectrum[n]
+            fluxes = np.hstack([fluxes, temp_spectrum])
+        # Append total spectrum    
+        fluxes = np.hstack([fluxes, neutron_spectrum])
+        # Append blank "zero" spectrum
+        fluxes = np.hstack([fluxes, np.zeros(num_n_groups)])
+    num_flux_blocks = num_mats * (num_n_groups + 2)
+    _gt_write_fluxin(fluxes, num_flux_blocks, fluxin_file)
+
+    # Write geom file
+    input_file = os.path.join(run_dir, "inp")
+    phtn_src_file = os.path.join(run_dir, "phtn_src")
+    _gt_write_inp(run_dir, data_dir, mats, num_n_groups, flux_magnitudes, 
+                  irr_times, decay_times, input_file, matlib_file,
+                  fluxin_file, phtn_src_file, num_p_groups, p_bins)
+
+    # Run ALARA
+    sub = subprocess.check_output(['alara', input_file], stderr=subprocess.STDOUT)
+    return phtn_src_file
+
+def calc_eta(data_dir, mats, neutron_spectrum, flux_magnitudes, irr_times,
+             decay_times, num_p_groups, p_bins, run_dir, clean):
+    """
+    Function that returns eta values (SNILB check result) for each material 
+    and each decay time
+    
+    Parameters
+    ----------
+    data_dir : str
+        Path to directory containing nuclib and fendl files
+    mats : list
+        List of PyNE material objects of materials in the geometry
+    neutron_spectrum : numpy array
+        Neutron energy spectrum (length is equal to number of n energy groups)
+    flux_magnitudes : list
+        Neutron flux normalization
+    irr_times : list
+        Irradiation times [s]
+    decay_times : list
+        Decay times [s]
+    num_p_groups : int
+        The number of photon energy groups for ALARA calculation
+    p_bins: numpy array
+       Photon energy bin bounds
+    run_dir: str
+        Path to write ALARA input and output files    
+    clean : bool
+        If True, remove run_dir
+        
+    Returns
+    ----------
+    eta : numpy array
+        eta value per photon group for each material listed.  
+        This is a 3D array [mat, decay_time, num_p_groups + 1]
+    """
+    num_n_groups = len(neutron_spectrum)
+    num_mats = len(mats)
+    num_decay_times = len(decay_times)
+    
+    # Run ALARA
+    if not os.path.exists(run_dir):
+        os.makedirs(run_dir)    
+    phtn_src_file = _gt_alara(data_dir, mats, neutron_spectrum, flux_magnitudes, 
+                              irr_times, decay_times, num_p_groups, p_bins, run_dir)
+    # Parse ALARA output
+    sup = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    tot = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    zero = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    with open(phtn_src_file, 'r') as f:
+        # Initiate a block number
+        i = 0
+        for line in f.readlines():
```

Comment:
couldn't this be a set of loops over m, n and dt, in which you throw away lines that don't match your "TOTAL" & "! shutdown" criteria?
It would be more readable.